src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java

```diff
@@ -314,6 +317,47 @@ public void castIntFieldToDoubleWithAliasJdbcFormatGroupByTest() {
         }
     }
 
+    @Test
+    public void castKeywordFieldToDatetimeWithoutAliasJdbcFormatTest() {
+        JSONObject response = executeJdbcRequest("SELECT CAST(date_keyword AS DATETIME) FROM "
+                + TestsConstants.TEST_INDEX_DATE);
+
+        String date_type_cast = "{\"name\":\"cast_date_keyword\",\"type\":\"date\"}";
+        assertEquals(response.getJSONArray("schema").get(0).toString(), date_type_cast);
+        String[] expectedOutput = new String[] {"Tue Aug 19 07:09:13 PDT 2014", "Wed Sep 25 02:04:13 PDT 2019"};
+        String[] response_str = new String[2];
+
+        for (int i = 0; i < response.getJSONArray("datarows").length(); ++i) {
+            response_str[i] = response.getJSONArray("datarows").getJSONArray(i).getString(0);
+        }
+        Arrays.sort(response_str);
+        Arrays.sort(expectedOutput);
+        for (int i = 0; i < response_str.length; ++i) {
+            Assert.assertThat(response_str[i], equalTo(expectedOutput[i]));
+        }
+    }
+
+    @Test
+    public void castKeywordFieldToDatetimeWithAliasJdbcFormatTest() {
+        JSONObject response = executeJdbcRequest("SELECT CAST(date_keyword AS DATETIME) AS test_alias FROM "
+                + TestsConstants.TEST_INDEX_DATE);
+
+        String date_type_cast = "{\"name\":\"test_alias\",\"type\":\"date\"}";
+        assertEquals(response.getJSONArray("schema").get(0).toString(), date_type_cast);
+        String[] expectedOutput = new String[] {"Wed Sep 25 02:04:13 PDT 2019", "Tue Aug 19 07:09:13 PDT 2014"};
+
+        String[] response_str = new String[2];
+
+        for (int i = 0; i < response.getJSONArray("datarows").length(); ++i) {
+            response_str[i] = response.getJSONArray("datarows").getJSONArray(i).getString(0);
+        }
+        Arrays.sort(response_str);
+        Arrays.sort(expectedOutput);
+        for (int i = 0; i < response_str.length; ++i) {
+            Assert.assertThat(response_str[i], equalTo(expectedOutput[i]));
+        }
```

Comment:
I would suggest to exempt the sorts and put these two loops together instead. That looks more concise and readable.

Reply:
The sort is there because the order of the JSONArray outputs are not consistent on each run.